universal-application-tool-0.0.1/test/repository/QuestionRepositoryTest.java

```diff
@@ -5,15 +5,55 @@
 import com.google.common.collect.ImmutableMap;
 import java.util.Locale;
 import java.util.Optional;
+import java.util.Set;
 import models.Question;
+import org.junit.Before;
 import org.junit.Test;
 import services.question.QuestionDefinition;
 
 public class QuestionRepositoryTest extends WithPostgresContainer {
 
+  private QuestionRepository repo;
+
+  @Before
+  public void setupQuestionRepository() {
+    repo = app.injector().instanceOf(QuestionRepository.class);
+  }
+
+  @Test
+  public void listQuestions_empty() {
+    assertThat(repo.listQuestions().toCompletableFuture().join()).isEmpty();
+  }
+
+  @Test
+  public void listQuestions() {
+    Question one = saveQuestion(1L);
+    Question two = saveQuestion(2L);
+
+    Set<Question> list = repo.listQuestions().toCompletableFuture().join();
+
+    assertThat(list).containsExactly(one, two);
+  }
+
+  @Test
+  public void lookupQuestion_returnsEmptyOptionalWhenQuestionNotFound() {
+    Optional<Question> found = repo.lookupQuestion(1L).toCompletableFuture().join();
+
+    assertThat(found).isEmpty();
+  }
+
+  @Test
+  public void lookupQuestion_findsCorrectQuestion() {
+    saveQuestion(1L);
+    Question existing = saveQuestion(20L);
+
+    Optional<Question> found = repo.lookupQuestion(existing.id).toCompletableFuture().join();
+
+    assertThat(found).hasValue(existing);
+  }
+
   @Test
-  public void createQuestion() {
-    QuestionRepository repo = app.injector().instanceOf(QuestionRepository.class);
+  public void insertQuestion() {
```

Comment:
(non-blocking  - because I think we probably have bad test names in several places...)

From our java testing best practices:

```
DON'T

Give tests names that say nothing beyond the name of the method being tested except in trivial cases.
```

While redundant, I think we should specify methodName_stateDetails_expectedOutput, which might mean `insertQuestion_insertsQuestion`.

Reply:
oh now i see it ends with `except in trivial cases`. duh.

Reply:
no worries, thanks for the reminder! agree that the lengthier method names help capture what the test intends to cover